### DISCUSSION TITLE
Restricting access to directories in s3 bucket

### DISCUSSION THREAD

**Discussion by pkk-code** [seaweedfs/seaweedfs#2551] [Q&A] [answered]
I am using static configuration like below and passing it to weed through -s3.config. 

    {
      "name": "restricted_to_directory",
      "credentials": [
        {
          "accessKey": "some_access_key",
          "secretKey": "some_secret_key"
        }
      ],
      "actions": [
        "Read:testnodejs/staging/*",
        "Write:testnodejs/staging/*",
        "List:testnodejs/staging/*"
      ]
    }

I do get access denied while accessing this directory using restricted user name.

Seaweed Log:

103 13:10:18 84031 auth_credentials.go:248] user name: restricted_to_directory actions: [Read:testnodejs/staging Write:testnodejs/staging List:testnodejs/staging], action: Read
I0103 13:10:18 84031 error_handler.go:85] status 403 application/xml: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied.</Message><Resource>/testnodejs/staging</Resource><RequestId>1641233418327046000</RequestId><Key>staging</Key><BucketName>testnodejs</BucketName></Error>

Can you guide me how to restrict access for a user to a certain directory in the bucket.

**Comment by chrislusf** [comment]
Added fix https://github.com/chrislusf/seaweedfs/commit/a7887166cfad779b693731dbf4fc1d678a2c81d0

**Reply by pkk-code** [reply]
It still fails.

Example:
aws --endpoint-url http://localhost:8333 s3 cp test.txt s3://testnodejs/staging/57b3bf77-5e8c-4603-b6fb-acadd6765088
upload failed: ./test.txt to s3://testnodejs/staging/57b3bf77-5e8c-4603-b6fb-acadd6765088 An error occurred (AccessDenied) when calling the PutObject operation: Access Denied.

Seaweed Log:
I0103 21:52:08 71172 auth_credentials.go:248] user name: restricted_to_directory actions: [Read:testnodejs/staging/* Write:testnodejs/staging/* List:testnodejs/staging/*], action: Write
I0103 21:52:08 71172 error_handler.go:85] status 403 application/xml: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied.</Message><Resource>/testnodejs/staging/57b3bf77-5e8c-4603-b6fb-acadd6765088</Resource><RequestId>1641264728756471000</RequestId><Key>staging/57b3bf77-5e8c-4603-b6fb-acadd6765088</Key><BucketName>testnodejs</BucketName></Error>

**Reply by chrislusf** [reply]
thanks for the quick verfication!

Added one more fix e76105e2abbbde00a9ae392bfdd5d20c7c706e89 for the fix. :)

**Comment by pkk-code** [comment, marked as answer]
Works now. Thanks for the quick turn around.
